agent/proxycfg/upstreams.go

```diff
@@ -301,12 +307,6 @@ func (s *handlerUpstreams) resetWatchesFromChain(
 		delete(snap.WatchedUpstreams[uid], targetID)
 		delete(snap.WatchedUpstreamEndpoints[uid], targetID)
 		cancelFn()
-
-		targetUID := NewUpstreamIDFromTargetID(targetID)
-		if targetUID.Peer != "" {
-			snap.PeerUpstreamEndpoints.CancelWatch(targetUID)
-			snap.UpstreamPeerTrustBundles.CancelWatch(targetUID.Peer)
-		}
```

Comment:
Is the reconcile handled above instead of in this specific case intentional?

I'm imagining we _should have_ been doing this in all cases before but were not?

Reply:
Good question!
This was canceling for a specific target and that is related to the bug. While in reality, because those watches could be shared by different targets, we would like to reconcile the list of watches every time to make sure that shared watches don't get canceled.